contrib/analysis/aggregate_data.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+
+import argparse
+
+# import multiprocessing
+import requests
+import time
+
+from datetime import datetime
+from dateutil import rrule
+from dateutil.relativedelta import relativedelta
+from pbench_combined_data import PbenchCombinedDataCollection
+
+from elasticsearch1 import Elasticsearch
+
+
+def _month_gen(now: datetime) -> str:
+    """Generate YYYY-MM stings from all the months, inclusively, between the
+    given start and end dates.
+    """
+    start = now - relativedelta(years=1)
```

Comment:
We should consider making the time window a parameter to this method, and default the value to one year using a declared constant.

Later we could consider adding a command line parameter to change the time window being considered.

Reply:
Perhaps rename parameter to be "beginning time" or "start time" instead of "now" so that it conveys the semantic of a time window of start / end.

Reply:
Renamed parameter to end_time, and added new param start_months_prior, added optional param to argparse for value of start_months_prior, defaults to 12